diff --git a/src/config.c b/src/config.c
--- a/src/config.c
+++ b/src/config.c
@@ -542,10 +542,10 @@
 		} else if (key_match("H1")) {
-			if ((ctx->device->init_packet_magic_header = strdup(value)) == NULL)
+			if ((ctx->device->init_packet_magic_header = strdup(value)) != NULL)
 				ctx->device->flags |= WGDEVICE_HAS_H1;
 		} else if (key_match("H2")) {
-			if ((ctx->device->response_packet_magic_header = strdup(value)) == NULL)
+			if ((ctx->device->response_packet_magic_header = strdup(value)) != NULL)
 				ctx->device->flags |= WGDEVICE_HAS_H2;
 		} else if (key_match("H3")) {
 			if ((ctx->device->underload_packet_magic_header = strdup(value)) != NULL)
 				ctx->device->flags |= WGDEVICE_HAS_H3;
 		} else if (key_match("H4")) {
